framework/include/loops/ComputeUserObjectsThread.h

```diff
@@ -118,10 +135,16 @@ groupUserObjects(TheWarehouse & w,
     }
 
     if ((obj->isParamValid("force_preaux") && obj->template getParam<bool>("force_preaux")) ||
-        aux_deps.count(obj->name()) > 0 || ic_deps.count(obj->name()) > 0)
+        ic_deps.count(obj->name()) > 0)
     {
-      w.update(obj, AttribPreAux(w, true));
+      pre_aux_dependencies[obj].insert(Moose::ALL);
```

Comment:
The Moose::ALL isn't meant to refer to execute_on flags - this makes the code confusing here.

Reply:
I was copying the functionality of the `AttribExecOns::isMatch(const Attribute & other) const` function where the flags are checked against `Moose::ALL` like so:
```c++
bool
AttribExecOns::isMatch(const Attribute & other) const
{
  auto a = dynamic_cast<const AttribExecOns *>(&other);
  if (!a || a->_vals.size() < 1)
    return false;
  auto cond = a->_vals[0];
  if (cond == Moose::ALL)
    return true;

  for (auto val : _vals)
    if (val == Moose::ALL || val == cond)
      return true;
  return false;
}
```
If this implementation will not work for `AttribPreAux` I can change it. Is there some other flag that indicates execute always?

Reply:
Also, I want to note that I removed the condition `aux_deps.count(obj->name()) > 0`. Maybe this bit of code is causing some confusion, it does not force the UO into `PRE_AUX` anytime there is a dependency. That is what the old code did, not the code in this MR.

The UO is still forced into PR_AUX if `ic_deps.count(obj->name()) > 0`. **Maybe this should be changed? Does it make sense to only force the UO into the EXEC_INITIAL group if an IC depends on it?**

Reply:
I need to think more about the UO grouping in this MR. As the code is currently in this MR, a UO can end up in more than one group, but that did not happen in the old code. I'll push a code update that fixes this

Reply:
I did change this in the most recent code update. I no longer use `MOOSE::ALL` for anything in `AttribPreAux` or `AttribExecOns`